loadtest/control/simulcontroller/actions.go

```diff
@@ -34,11 +37,35 @@ func (c *SimulController) reload(full bool) control.UserActionResponse {
 		if err != nil {
 			return control.UserActionResponse{Err: control.NewUserError(err)}
 		}
-
+		c.user.ClearUserData()
 		c.connect()
 	}
 
-	return control.Reload(c.user)
+	if resp := control.Reload(c.user); resp.Err != nil {
+		return resp
+	} else {
+		c.status <- c.newInfoStatus(resp.Info)
+	}
+
+	team, err := c.user.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team == nil {
+		// If the current team is not set we switch to a random one.
+		return c.switchTeam(c.user)
+	}
+
+	return loadTeam(c.user, team)
+}
+
+func (c *SimulController) login() control.UserActionResponse {
+	if resp := control.Login(c.user); resp.Err != nil {
+		return resp
+	} else {
+		c.connect()
+		go c.wsEventHandler()
```

Comment:
One concern regarding this: the wsEventHandler chan is only closed once when the controller is stopped.

In simplecontroller, we were just calling it once in the Run method. But now we are going to call it multiple times ? This will keep other goroutines running. Unless I have missed something.

Reply:
You are correct. Need to fix this.

Reply:
It should be okay now. I made the goroutine creation happen on the initial login only.